lisa/features/password_extension.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa.feature import Feature
+
+FEATURE_NAME_PASSWORD_EXTENSION = "PasswordExtension"
+
+
+class PasswordExtension(Feature):
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_PASSWORD_EXTENSION
+
+    @classmethod
+    def can_disable(cls) -> bool:
+        raise NotImplementedError()
+
+    def enabled(self) -> bool:
+        raise NotImplementedError()
+
+    def is_supported(self) -> bool:
+        raise NotImplementedError()
+
+    def reset_password(self, username: str, password: str) -> None:
+        raise NotImplementedError()
+
+    def generate_password(self) -> str:
```

Comment:
`generate_password` should be implement here, because it's shared.

Reply:
Deleted this function.

Reply:
use cherry pick to pick the commit, instead of merging it into existing commits. You can reorder commits, to make sure the committed order is valid.